### ISSUE TITLE
Add Spack Developer Workflow for E4S 21.05

### ISSUE THREAD

**Issue by shahzebsiddiqui** [buildtesters/buildtest-nersc#88]
@wspear 
Similar to issue https://github.com/buildtesters/buildtest-nersc/issues/84 we would like to get a spack developer workflow for this stack https://docs.nersc.gov/applications/e4s/cori/21.05/ 

We need to contribute a test in https://github.com/buildtesters/buildtest-nersc/tree/devel/buildspecs/apps/e4s that tests the user developer section used to install packages.

**Comment by shahzebsiddiqui** [comment]
That's for testing this out. Yeah I think this stack requires rebuild if you want to consult with Luke. We can start the rebuild after finishing up 22.05 and hold off on this test for time being since it's clearly not working and documentation is not reproducible.

Since this stack provides a build cache we may omit this going forward since it is useless as system changes the buildcache can't be used

**Comment by wspear** [comment]
A test deleting the ~/.spack directory makes me nervous. Would it be ok to move .spack to .spack-<timestamp> instead, just in case somebody else runs this and unexpectedly loses their user configuration?

**Comment by shahzebsiddiqui** [comment]
Yeah now that i think about this we are testing the user documentation and since we don't document this `i.e rm -rf ~/.spack` in the workflow then i think we shouldn't include this just for fixing the test. Problem with `mv` statement is you can get error when the source directory doesn't exist.

**Comment by wspear** [comment]
Can this be closed with https://software.nersc.gov/NERSC/buildtest-nersc/-/merge_requests/60 ?
